sds/src/scanner/mod.rs

```diff
@@ -1196,39 +1206,50 @@ mod test {
                 replacement_type: crate::ReplacementType::None,
                 start_index: 0,
                 end_index_exclusive: 3,
-                shift_offset: 0
+                shift_offset: 0,
             }
         );
     }
 
     #[test]
     fn should_skip_match_when_present_in_excluded_matches() {
-        // If 2 matches have the same mutation and same start, the longer one is taken
+        use metrics::{Key, KeyName, SharedString};
+        use metrics_util::debugging::DebugValue;
+        use metrics_util::CompositeKey;
+        use metrics_util::MetricKind::Counter;
 
-        let rule_0 = RegexRuleConfig::builder("b.*".to_owned())
-            .scope(Scope::exclude(vec![Path::from(vec![PathSegment::Field(
-                "test".into(),
-            )])]))
-            .match_action(MatchAction::Redact {
-                replacement: "[scrub]".to_string(),
-            })
-            .build();
-
-        let scanner = Scanner::new(&[rule_0]).unwrap();
+        let recorder = DebuggingRecorder::new();
+        let snapshotter = recorder.snapshotter();
 
-        let mut content = SimpleEvent::Map(BTreeMap::from([
-            (
-                "a-match".to_string(),
-                SimpleEvent::String("bcdef".to_string()),
-            ),
-            (
-                "z-match".to_string(),
-                SimpleEvent::String("bcdef".to_string()),
-            ),
-            ("test".to_string(), SimpleEvent::String("bcdef".to_string())),
-        ]));
+        let matches = metrics::with_local_recorder(&recorder, || {
```

Comment:
In general it's better to create a new test for specifically testing metrics. This unit test was already a slightly more complicated one where the order of the fields matters. It's probably better to create a new one with a more simple test to isolate the test to just the metrics testing as much as possible

Reply:
Yes definitely! typing...

Reply:
[Refactor tests](https://github.com/DataDog/dd-sensitive-data-scanner/pull/40/commits/3e65fc841e51866c89acec7e08121c0440cefca3)